crates/typst-library/src/text/quotes.rs

```diff
@@ -228,14 +277,117 @@ impl<'s> Quotes<'s> {
     }
 }
 
-impl Default for Quotes<'_> {
-    /// Returns the english quotes as default.
-    fn default() -> Self {
-        Self {
-            single_open: "‘",
-            single_close: "’",
-            double_open: "“",
-            double_close: "”",
-        }
+/// An opening and closing quote.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct QuoteSet {
+    open: EcoString,
+    close: EcoString,
+}
+
+cast! {
+    QuoteSet,
+    self => {
+        [self.open, self.close]
+            .into_iter()
+            .map(IntoValue::into_value)
+            .collect::<Array>()
+            .into_value()
+    },
+    value: Array => {
+        let [open, close] = array_to_set(value)?;
+        Self { open, close }
+    },
+    value: Str => {
+        let [open, close] = str_to_set(value.as_str())?;
+        Self { open, close }
+    },
+}
+
+fn str_to_set(value: &str) -> StrResult<[EcoString; 2]> {
+    let mut iter = value.graphemes(true);
+    let open = iter
+        .next()
+        .ok_or_else(|| eco_format!("expected 2 characters, got 0 characters"))?;
```

Comment:
you're using "characters" in errors & docs where it's actually graphemes. I'm on the fence of whether using "graphemes" is too technical, but it's not really accurate currently.

Reply:
I know, this was something I was unsure about. The average user thinks of these as visible "characters". Perhaps it should be added as a note that it actually does use graphemes?